extensions/sql-migration/src/api/utils.ts

```diff
@@ -600,6 +600,53 @@ export async function getVirtualMachines(account?: Account, subscription?: azure
 	return virtualMachines;
 }
 
+export async function getVirtualMachinesDropdownValues(virtualMachines: azure.SqlVMServer[], location: azureResource.AzureLocation, resourceGroup: azureResource.AzureResourceResourceGroup, account: Account, subscription: azureResource.AzureResourceSubscription): Promise<CategoryValue[]> {
+	let virtualMachinesValues: CategoryValue[] = [];
+	if (location && resourceGroup) {
+		for (const virtualMachine of virtualMachines) {
+			if (virtualMachine.location.toLowerCase() === location.name.toLowerCase() && azure.getResourceGroupFromId(virtualMachine.id).toLowerCase() === resourceGroup.name.toLowerCase()) {
+				let virtualMachineValue: CategoryValue;
+
+				// 1) check if VM is on by querying underlying compute resource's instance view
+				let vmInstanceView = await azure.getVMInstanceView(virtualMachine, account, subscription);
+				if (!vmInstanceView.statuses.some(status => status.code == "PowerState/running")) {
```

Comment:
should we do case insensitive compare here?

Reply:
Second on this.
Case-insensitive comparing for "PowerState/running" and "Full" is safer.

Reply:
Good call, updated